Proof doc (#8)
scalar-labs/scalardl: docs/how-to-use-proof.md

# A Guide on How to Use Asset Proofs in Scalar DL

This document sets out some guidelines for using Asset Proofs in Scalar DL.

> ## What is an Asset Proof in Scalar DL ?

Author: ```suggestion
## What is an Asset Proof in Scalar DL?
```

> An Asset Proof in Scalar DL is a set of information about an asset record that is created along with the asset. It is composed of the following records so that it can be treated as evidence of creation of an asset record by a ledger.

Author: ```suggestion
An Asset Proof in Scalar DL is a set of information about an asset record that is created along with the asset. It is composed of the following records so that it can be treated as evidence of the creation of an asset record by a ledger.
```

> - ID of an asset record

Author: ```suggestion
- The ID of an asset record
```

- Age of the asset record

> - Nonce of the execution request that creats the asset record

Author: ```suggestion
- Nonce of the execution request that creates the asset record
```

> - Cryptographic hash of the asset record

Author: ```suggestion
- A cryptographic hash of the asset record
```

> - Digita sigunature of the above four records 

Author: ```suggestion
- The digital signature of the above four records 
```

> ## The Benefits of Asset Proof

Author: ```suggestion
## The Benefits of Asset Proofs
```

> Since Asset Proofs are evidence at the time of exucution by a ledger, it is hard for the leger to make a lie (tamper data) after the evidence is created because the proofs and the ledger states would be inconsistent.

Author: ```suggestion
Since Asset Proofs are evidence at the time of execution by a ledger, it is hard for the leger to make a lie (tamper data) after the evidence is created because the proofs and the ledger states would be inconsistent.
```

Thus, using and managing Asset Proofs appropriately could reduce the risk of tampering of data, so it could make Scalar DL more tamper-evident without creating additional computational resources such as ordering components.
However, the ledger can make a lie from the beginning and it makes the proofs insufficient to detect the tampering.

> ## How to use Asset Proof

Author: ```suggestion
## How to use Asset Proofs
```

> You can get Asset Proofs from the result of `executeContract` in the SDKs. Please check [the documents](https://github.com/scalar-labs/scalardl#client-sdks) of client SDKs for more detail.

Author: ```suggestion
You can get Asset Proofs from the result of the `executeContract` method of the SDKs. Please check [the documents](https://github.com/scalar-labs/scalardl#client-sdks) of client SDKs for more detail.
```

> It is recommended to store the proofs outside of a domain in which the ledger is managed so that malicious activities in one domain can be detectd by the other domain.

Author: ```suggestion
It is recommended to store the proofs outside of a domain in which the ledger is managed so that malicious activities in one domain can be detected by the other domain.
```

> It is worth considering storing proofs in cloud storages for ease of management.

Author: ```suggestion
It is worth considering storing proofs in cloud storage for ease of management.
```

# A Guide on How to Use Asset Proofs in Scalar DL

This document sets out some guidelines for using Asset Proofs in Scalar DL.

## What is an Asset Proof in Scalar DL?

An Asset Proof in Scalar DL is a set of information about an asset record that is created along with the asset record. It is composed of the following items so that it can be treated as evidence of the creation of an asset record by a ledger.

- The ID of an asset record
- Age of the asset record
- Nonce of the execution request that creates the asset record
- A cryptographic hash of the asset record
- The digital signature of the above four records 

## The Benefits of Asset Proofs

> Since Asset Proofs are evidence at the time of execution by a ledger, it is hard for the leger to make a lie (tamper data) after the evidence is created because the proofs and the ledger states would be inconsistent.

Author: ```suggestion
Since Asset Proofs are evidence at the time of execution by a ledger, it is hard for the leger to tamper data after the evidence is created because the proofs and the ledger states would be inconsistent.
```

> Thus, using and managing Asset Proofs appropriately could reduce the risk of tampering of data, so it could make Scalar DL more tamper-evident without creating additional computational resources such as ordering components.

Reviewer: What does "using and managing Asset Proofs" mean in this document?
I'm not sure why "managing" is with "using" because their subjects are different.
I think a user or validator can use AssetProofs since Scalar DL manages them.

Author: Good point!
From my perspective, the user-side or client-side not only uses proofs but also can manage them like storing them in client-side storage for later use. In contract, the ledger just creates them and doesn't manage them so it doesn't store them, for example. That's what I meant for `manage` to specify clients can/should do more than `use`.
But `manage` might be a too blur word. If you feel it is confusing, maybe we can say like this to imply that client can do more than just `using`. How does it sound?

`using and managing` -> `making use of`

Reviewer: Thank you for your reply! It makes sense 👍 
`making use of` looks better.

Author: OK! I updated it with 6163a61.

> Thus, using and managing Asset Proofs appropriately could reduce the risk of tampering of data, so it could make Scalar DL more tamper-evident without creating additional computational resources such as ordering components.

Author: ```suggestion
Thus, making use of Asset Proofs appropriately could reduce the risk of tampering of data, so it could make Scalar DL more tamper-evident without creating additional computational resources such as ordering components.
```

> However, the ledger can make a lie from the beginning and it makes the proofs insufficient to detect the tampering.

Author: ```suggestion
However, the ledger can tamper data from the beginning and it makes the proofs insufficient to detect the tampering.
```

## How to use Asset Proofs

You can get Asset Proofs from the result of the `executeContract` method of the SDKs. Please check [the documents](https://github.com/scalar-labs/scalardl#client-sdks) of client SDKs for more detail.

> The proofs can be validated if it is not tampered and it is from the ledger by verifying the signature.

Author: ```suggestion
A proof can be validated if it has not tampered and it is from the ledger by verifying the signature.
```
